pkg/services/query/query.go

```diff
@@ -90,18 +91,31 @@ func (s *Service) QueryDataMultipleSources(ctx context.Context, user *user.Signe
 	} else {
 		resp := backend.NewQueryDataResponse()
 
-		// create new reqDTO with only the queries for that datasource
+		g, ctx := errgroup.WithContext(ctx)
+		results := make([]backend.Responses, len(byDataSource))
+
 		for _, queries := range byDataSource {
-			subDTO := reqDTO.CloneWithQueries(queries)
+			dataSourceQueries := queries
+			g.Go(func() error {
+				subDTO := reqDTO.CloneWithQueries(dataSourceQueries)
 
-			subResp, err := s.QueryData(ctx, user, skipCache, subDTO, handleExpressions)
+				subResp, err := s.QueryData(ctx, user, skipCache, subDTO, handleExpressions)
 
-			if err != nil {
-				return nil, err
-			}
+				if err == nil {
+					results = append(results, subResp.Responses)
+				}
+
+				return err
+			})
+		}
+
+		if err := g.Wait(); err != nil {
```

Comment:
looks good!  but still sad this is not part of the regular query endpoint.

The regular one is missing mixed support because it was not supported... not because it should not happen!

Reply:
@ryantxu I've added it to our pipeline : ) We'll do as a separate PR.